src/cider/nrepl/middleware/util/instrument.clj

```diff
@@ -164,12 +164,16 @@
 (defn- contains-recur?
   "Return true if form is not a `loop` or a `fn` and a `recur` is found in it."
   [form]
-  (if (seq? form)
-    (case (first form)
-      recur true
-      loop* false
-      fn*   false
-      (some contains-recur? (rest form)))))
+  (cond (seq? form) (case (first form)
+                      recur true
+                      loop* false
+                      fn*   false
+                      (some contains-recur? (rest form)))
+        ;; case* expands the non-default branches into a map
```

Comment:
I think this comment should probably be expanded a bit for posterity's sake. Perhaps you can also add here your remark that we expect `recur` to appear in a map/vector only as the result of a `case` expansion.